diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +124,169 @@ class LineLength : Rule("line-length") {
+                else -> parent = parent.treeParent
+            }
+        } while (parent.treeParent != null)
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        val leftOffset = positionByOffset(wrongNode.startOffset).second
+        val indexLastSpace = wrongNode.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        if (indexLastSpace == -1)
+            return
+        val nodeText = "//${wrongNode.text.substring(indexLastSpace, wrongNode.text.length)}"
+        wrongNode.treeParent.run {
+            addChild(LeafPsiElement(EOL_COMMENT, wrongNode.text.substring(0, indexLastSpace)), wrongNode)
+            addChild(PsiWhiteSpaceImpl("\n"), wrongNode)
+            addChild(LeafPsiElement(EOL_COMMENT, nodeText), wrongNode)
+            removeChild(wrongNode)
+        }
+    }
+
+    /**
+     * This method fix too long binary expression: split after OPERATION_REFERENCE closest to max length
+     * @param wrongNode node that should be split
+     * @param configuration max length configuration
+     * @param leftInitOffset is a left offset, it need when we split binary expression in right value case
+     * @param isProperty if wrongNode is a property
+     *
+     * In this method we collect all binary expression in correct order and then
+     * we collect their if their length less then max.
+     */
+    private fun fixLongBinaryExpression(wrongNode: ASTNode, configuration: LineLengthConfiguration,
+                                        leftInitOffset: Int = -1, isParenthesized: Boolean = false) {
+        var leftOffset = if (leftInitOffset < 0) {
+            positionByOffset(wrongNode.firstChildNode.startOffset).second
+        } else leftInitOffset
+        val binList = mutableListOf<ASTNode>()
+        if (isParenthesized)
+            dfsForProperty(wrongNode, binList)
+        else
+            searchBinaryExpression(wrongNode, binList)
+        var binaryText = ""
+        binList.forEachIndexed { index, astNode ->
+            if (index == 0) {
+                binaryText += astNode.treeParent.prevSibling { it.elementType == LPAR }?.text ?: ""
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+            } else {
+                binaryText += astNode.treeParent.prevSibling { it.elementType == LPAR }?.text ?: ""
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+                binaryText += astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!.text
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE }?.text ?: ""
+                binaryText += astNode.treeParent.nextSibling { it.elementType == RPAR }?.text ?: ""
```

Comment:
Shouldn't this line and also with `LPAR` be needed only if `treeParent.elementType == PARENTHESIZED`? If so, please add comment about it. If you can't create example where it is not true (I can't), additional check could be redundant, but comment will be helpful.